src/main/java/fr/xephi/authme/service/BungeeService.java

```diff
@@ -1,69 +1,191 @@
+        }
+        if (!messenger.isIncomingChannelRegistered(plugin, "BungeeCord")) {
+            messenger.registerIncomingPluginChannel(plugin, "BungeeCord", this);
+        }
+    }
+
+    private void sendBungeecordMessage(String... data) {
+        ByteArrayDataOutput out = ByteStreams.newDataOutput();
+        for(String element : data) {
+            out.writeUTF(element);
+        }
+        service.sendPluginMessage("BungeeCord", out.toByteArray());
+    }
+
     /**
      * Send a player to a specified server. If no server is configured, this will
      * do nothing.
      *
      * @param player The player to send.
      */
-    public void connectPlayer(Player player) {
-        if (!isEnabled || bungeeServer.isEmpty()) {
+    public void connectPlayerOnLogin(Player player) {
+        if (!isEnabled || destinationServerOnLogin.isEmpty()) {
             return;
         }
 
-        service.scheduleSyncDelayedTask(new Runnable() {
-            @Override
-            public void run() {
-                ByteArrayDataOutput out = ByteStreams.newDataOutput();
-                out.writeUTF("Connect");
-                out.writeUTF(bungeeServer);
-                player.sendPluginMessage(plugin, "BungeeCord", out.toByteArray());
-            }
-        }, 20L);
+        service.scheduleSyncDelayedTask(() ->
+            sendBungeecordMessage("Connect", player.getName(), destinationServerOnLogin), 20L);
+    }
+
+    private void sendAuthMeBungeecordMessage(String type, String... data) {
+        if(!isEnabled) {
+            return;
+        }
+
+        List<String> dataList = Arrays.asList(data);
+        dataList.add(0, "AuthMe");
+        dataList.add(1, type);
+        sendBungeecordMessage(dataList.toArray(new String[dataList.size()]));
+    }
+
+    public void sendLogin(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.LOGIN, name.toLowerCase());
+    }
+
+    public void sendLogout(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.LOGOUT, name.toLowerCase());
+    }
+
+    public void sendRegister(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REGISTER, name.toLowerCase());
+    }
+
+    public void sendUnregister(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.UNREGISTER, name.toLowerCase());
+    }
+
+    public void sendRefreshPassword(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REFRESH_PASSWORD, name.toLowerCase());
+    }
+
+    public void sendRefreshSession(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REFRESH_SESSION, name.toLowerCase());
+    }
+
+    public void sendRefreshQuitLoc(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REFRESH_QUITLOC, name.toLowerCase());
+    }
+
+    public void sendRefreshEmail(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REFRESH_EMAIL, name.toLowerCase());
```

Comment:
Given that you're sending `name.toLowerCase()` to the same method from twelve different places, you could just use `toLowerCase()` inside `sendAuthMeBungeecordMessage`

Reply:
Right xD